src/Sets/SparsePolynomialZonotope.jl

```diff
@@ -207,6 +207,42 @@ indexvector(P::SPZ) = P.idx
 Returns a collection of n unique identifiers (intergers 1, …, n).
 """
 uniqueID(n::Int) = 1:n
+
+"""
+    mergeID(P1::SparsePolynomialZonotope, P2::SparsePolynomialZonotope)
+
+Returns two polynomial zonotopes equivalent to the correspondent input but with the same
+index vector.
+
+### Input
+
+- `P1` -- sparse polynomial zonotope
+- `P2` -- sparse polynomial zonotope
+
+### Output
+
+two polynomial zonotopes with the same index vector
+"""
+function mergeID(P1::SparsePolynomialZonotope, P2::SparsePolynomialZonotope)
+
+    idx1 = indexvector(P1)
+    idx2 = indexvector(P2)
+    idx1 == idx2 && return (P1, P2)
+
+    H = setdiff(idx2, idx1)
+    idx_new = vcat(idx1, H)
+    E1 = vcat(expmat(P1), zeros(Int, length(H), ndependentgens(P1)))
+    E2 = zeros(Int, length(idx1) + length(H), ndependentgens(P2))
+    E2_old = expmat(P2)
+    @inbounds for (i, idx) in enumerate(idx_new)
+        j = findfirst(isequal(idx), idx2)
+        !isnothing(j) && (E2[i, :] .= E2_old[j, :])
+    end
+    P1_new = SparsePolynomialZonotope(center(P1), dependent_genmat(P1), independent_genmat(P1), E1, idx_new)
```

Comment:
unrelated but i find it cleaner to read (also to find by `genmat[TAB]`
```julia
 SparsePolynomialZonotope(center(P1), genmat_dep(P1), genmat_indep(P1), E1, idx_new)
```
see https://github.com/JuliaReach/LazySets.jl/issues/3031

Reply:
should I fix that first and then rebase the open PRs or implement the renaming after these PRs are merged?

Reply:
we can do that later